src/pyhf/contrib/viz/brazil.py

```diff
@@ -24,27 +24,205 @@ def plot_results(ax, mutests, tests, test_size=0.05):
 
     Args:
-        ax (`matplotlib.axes.Axes`): The matplotlib axis object to plot on.
+        ax (:obj:`matplotlib.axes.Axes`): The matplotlib axis object to plot on.
         mutests (:obj:`list` or :obj:`array`): The values of the POI where the
           hypothesis tests were performed.
-        tests (:obj:`list` or :obj:`array`): The :math:`\\mathrm{CL}_{s}` values
+        tests (:obj:`list` or :obj:`array`): The :math:`\mathrm{CL}_{s}` values
           from the hypothesis tests.
-        test_size (:obj:`float`): The size, :math:`\\alpha`, of the test.
+        test_size (:obj:`float`): The size, :math:`\alpha`, of the test.
+
+    Returns:
+        :obj:`list`: The list of lists and tuples of :obj:`matplotlib.artist` drawn.
     """
     cls_obs = np.array([test[0] for test in tests]).flatten()
     cls_exp = [np.array([test[1][i] for test in tests]).flatten() for i in range(5)]
-    ax.plot(mutests, cls_obs, c='black')
-    for idx, color in zip(range(5), 5 * ['black']):
-        ax.plot(
-            mutests, cls_exp[idx], c=color, linestyle='dotted' if idx != 2 else 'dashed'
+
+    line_color = kwargs.pop("color", "black")
+    axis_artists = [
+        ax.plot(mutests, cls_obs, color=line_color, label=r"$\mathrm{CL}_{s}$")
+    ]
+
+    for idx, color in zip(range(5), 5 * [line_color]):
+        _cls_exp_line = ax.plot(
+            mutests,
+            cls_exp[idx],
+            color=color,
+            linestyle="dotted" if idx != 2 else "dashed",
+            label=None if idx != 2 else r"$\mathrm{CL}_{s,\mathrm{exp}}$",
         )
-    ax.fill_between(mutests, cls_exp[0], cls_exp[-1], facecolor='yellow')
-    ax.fill_between(mutests, cls_exp[1], cls_exp[-2], facecolor='green')
-    ax.plot(mutests, [test_size] * len(mutests), c='red')
+        axis_artists.append(_cls_exp_line)
+    one_sigma_band = ax.fill_between(
+        mutests,
+        cls_exp[0],
+        cls_exp[-1],
+        facecolor="yellow",
+        label=r"$\pm2\sigma$ $\mathrm{CL}_{s,\mathrm{exp}}$",
+    )
+    two_sigma_band = ax.fill_between(
+        mutests,
+        cls_exp[1],
+        cls_exp[-2],
+        facecolor="green",
+        label=r"$\pm1\sigma$ $\mathrm{CL}_{s,\mathrm{exp}}$",
+    )
+
+    test_size_color = kwargs.pop("test_size_color", "red")
+    test_size_linestyle = kwargs.pop("test_size_linestyle", "solid")
+    test_size_line = ax.plot(
+        mutests,
+        [test_size] * len(mutests),
+        color=test_size_color,
+        linestyle=test_size_linestyle,
+        label=rf"$\alpha={test_size}$",
+    )
     ax.set_ylim(0, 1)
 
-    ax.set_xlabel(r"$\mu$ (POI)")
-    ax.set_ylabel(r"$\mathrm{CL}_{s}$")
+    x_label = kwargs.pop("xlabel", r"$\mu$ (POI)")
+    y_label = kwargs.pop("ylabel", r"$\mathrm{CL}_{s}$")
+    ax.set_xlabel(x_label)
+    ax.set_ylabel(y_label)
+
+    axis_artists.extend((one_sigma_band, two_sigma_band, test_size_line))
+    return axis_artists
```

Comment:
After learning more about how to be matplotlib like from a PR on `hist` I changed this so that it now returns a list of the artists like a normal `maplotlib` axis function would.